### PR TITLE
Merge UCL/STIR/master 5Apr2022

### PR DESCRIPTION
**Pull request by robbietuk**
@KrisThielemans The merge conflicts were tough, please check it over (or repeat it). I will try to review the parts I found hardest and my have gotten wrong

### CODE REVIEW

File: src/include/stir/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.h [hunk 1]
```diff
@@ -75,9 +75,16 @@ class BackProjectorByBinParallelproj :
...
+  void set_defaults();
+
   /// Set verbosity
   void set_verbosity(const bool verbosity) { _cuda_verbosity = verbosity; }
 
+   // set/get number of gpu chunks to use
+   void set_num_gpu_chunks(int num_gpu_chunks) {_num_gpu_chunks = num_gpu_chunks; }
+   int  get_num_gpu_chunks() { return _num_gpu_chunks; }
+
   BackProjectorByBinParallelproj* clone() const override;
```
Comment (robbietuk) [inline src/include/stir/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.h:88]: I had a merge conflict here. `BackProjectorByBinParallelproj* clone() const override;` does not exists on UCL/STIR but was on `tof_sino_UCL`

File: src/buildblock/Scanner.cxx [hunk 2]
```diff
@@ -798,6 +798,53 @@ Scanner::
...
+             transaxial_block_spacing_v,
+             crystal_map_file_name_v);
+}
+
+
+void
+Scanner::
+set_params(Type type_v,const list<string>& list_of_names_v, 
+           int num_rings_v, 
+           int max_num_non_arccorrected_bins_v,
```
Comment (robbietuk) [inline src/buildblock/Scanner.cxx]: This looked weird at the time but might be fine

**Review by robbietuk — COMMENTED** [review]
Some merge conflicts that were difficult?

File: src/buildblock/Scanner.cxx (hunk 2 above)
Comment (robbietuk) [reply src/buildblock/Scanner.cxx]: There is a build in here. There appears to be a lack of TOF variables? I am too tired to fix this tonight, Ill check again tomorrow

Comment (KrisThielemans) [reply src/buildblock/Scanner.cxx]: you re-inserted the `set_params()` function that was removed, so I've got rid of it again

File: src/include/stir/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.h (hunk 1 above)
Comment (KrisThielemans) [reply src/include/stir/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.h:88]: sure. this is fine

**Comment by KrisThielemans** [comment]
This seems fine to me now. Obviously, the tests will fail due to #26

**Comment by robbietuk** [comment]
Looks great to me. I think it is ready.

**Pull request merged**